src/activities/bargame/Bargame.qml

```diff
@@ -0,0 +1,341 @@
+                    width: rootWindow.width / (15 + Activity.ballSizeFactor[Activity.sublevel - 1])
+                }
+            }
+        }
+
+        // Lower green balls sample
+        Grid {
+            id: greenBalls
+            x: rootWindow.width / 5
+            y: rootWindow.height / 1.2
+            rows: 1
+            columns: 4
+            Repeater {
+                model: greenBalls.columns
+                Image {
+                    id: greenBall
+                    source: Activity.url + "green_ball.svg"
+                    height: rootWindow.height / (8 + Activity.ballSizeFactor[Activity.sublevel - 1])
+                    width: rootWindow.width / (15 + Activity.ballSizeFactor[Activity.sublevel - 1])
+                }
+            }
+        }
+
+        // Box row
+        Item {
+            id: boxModel
+            // The empty boxes grid
+            Grid {
+                id: boxes
+                rows: 1
+                columns: 15
+                x: 0
+                y: rootWindow.height / 1.4
+                Repeater {
+                    id: startCase
+                    model: boxes.columns
+                    Image {
+                        id: greenCase
+                        source: Activity.url + ((index == boxes.columns - 1) ? "case_last.svg" : "case.svg")
+                        height: rootWindow.height / (9 + (Activity.sublevel - 1) * Activity.elementSizeFactor[Activity.sublevel - 1])
+                        width: rootWindow.width / (15 + (Activity.sublevel - 1) * Activity.elementSizeFactor[Activity.sublevel - 1])
+                        visible: true
+                        // Numbering label
+                        GCText {
+                            text: index + 1
+                            fontSize: smallSize
+                            font.bold: true
+                            visible: ((index + 1) % 5 == 0 && index > 0) ? true : false
+                            anchors {
+                                horizontalCenter: parent.horizontalCenter
+                                bottom: parent.top
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Hidden Answer Balls
+            Item {
+                id: answerBalls
+                // All green balls placement
+                Grid {
+                    id: answerBallsPlacement
+                    x: rootWindow.width / 2000
+                    y: rootWindow.height / 1.4
+                    columns: 15
+                    rows: 1
+                    Repeater {
+                        model: answerBallsPlacement.columns
+                        Image {
+                            id: greenBall
+                            source: Activity.url + "green_ball.svg"
+                            height: rootWindow.height / (9 + (Activity.sublevel - 1) * Activity.elementSizeFactor[Activity.sublevel - 1])
```

Comment:
A readonly variable would create a problem with the dynamic size change of the elements during the gameplay which we might not want to happen. I'd rather try to shorten it a bit